code/controllers/subsystem/air.dm

```diff
@@ -603,10 +603,31 @@ SUBSYSTEM_DEF(air)
 			EG.dismantle()
 			CHECK_TICK
 
+		log_active_turfs() // invoke this here so we can count the time it takes to run this proc as "wasted time", quite simple honestly.
+
 		var/msg = "HEY! LISTEN! [DisplayTimeText(world.timeofday - timer, 0.00001)] were wasted processing [starting_ats] turf(s) (connected to [ending_ats - starting_ats] other turfs) with atmos differences at round start."
 		to_chat(world, span_boldannounce("[msg]"))
 		warning(msg)
 
+/// Logs all active turfs at roundstart to the mapping log so it can be readily accessed.
+/datum/controller/subsystem/air/proc/log_active_turfs()
+// sadly this has to be here because we can't realistically expect that all active turfs will be resolved in every possible situation when running through CI.
+// In an ideal world, we would have absolutely zero active turfs 99.99% of the time, but that's not the case. `log_mapping()` during world initialize triggers a CI fail.
+#ifdef UNIT_TESTS
+	return
```

Comment:
you're not comfortable with requiring no active turfs? too random?

Reply:
requiring active turfs would be the dream goal at the end of the day but as long as we have random ruin placement and modular mapping this would be the most random thing to ever break in the history of "random things that break sometimes"

Reply:
one thing that i was considering doing a few months earlier was having a "maximum allowable time" to resolve active turfs, and anything above that would be considered a failure so it would remain visible since it was being newly introduced in a PR rather than already being extant on prod- but i wasn't sure if that would be useful at all

Reply:
> requiring complete removal of active turfs would be the dream goal at the end of the day but as long as we have random ruin placement and modular mapping this would be the most random thing to ever break in the history of "random things that break sometimes". we still don't even have enough data to prune out the ones that persist to this day

wouldn't this pr help with that data? I'm not familiar with how widespread the problem is but combined with the flakey issues bot this seems like something we could wrangle

Reply:
it would yes, but we can't really jump the gun and make it a CI step without the data. i wish i could run this a bunch of times over all maps running every potential ruin placement but i don't have the time for that unfortunately

Reply:
I agree with san, unit testing and CI should be very false positive adverse (i consider any ci failure on a pr that wasn't caused by the changes in that pr to be a false positive) because that builds up into a death by a thousand cuts.

Reply:
i included a lot more to this PR for data collection for a potential scenario in the future where we could enable CI failures piecemeal.

remaining realistic, it is likely that this data goes nowhere, but at least in the future if someone much more ironwilled on rooting these out than I can will be able to justify their position on linting against every single active turf quite easily